cdap-master/src/main/java/co/cask/cdap/master/startup/HBaseCheck.java

```diff
@@ -66,5 +70,41 @@ public void run() {
           "Please check that HBase is running and that the correct HBase configuration (hbase-site.xml) " +
           "and libraries are included in the CDAP master classpath.", e);
     }
+
+    if (hConf.getBoolean("hbase.security.authorization", false)) {
+      if (cConf.getBoolean(TxConstants.TransactionPruning.PRUNE_ENABLE)) {
+        LOG.info("Transaction pruning is enabled. Checking global admin privileges for cdap.");
+        try {
+          boolean isGlobalAdmin = hBaseTableUtil.isGlobalAdmin(hConf);
+          LOG.info("Global admin privileges check status: {}", isGlobalAdmin);
+          if (isGlobalAdmin) {
+            return;
+          }
+          // if global admin was false then depend on the TX_PRUNE_ACL_CHECK value
+          if (cConf.getBoolean(Constants.Startup.TX_PRUNE_ACL_CHECK, false)) {
+            LOG.info("Found {} to be set to true. Continuing with cdap master startup even though global admin check " +
+                       "returned false", Constants.Startup.TX_PRUNE_ACL_CHECK);
+            return;
+          }
+          StringBuilder builder = new StringBuilder("Transaction pruning is enabled and cdap does not have global " +
+                                                      "admin privileges in HBase. Global admin privileges for cdap " +
+                                                      "is required for transaction pruning. " +
+                                                      "Either disable transaction pruning or grant global admin " +
+                                                      "privilege to cdap in HBase.");
```

Comment:
Or at your own risk override this check by setting TX_PRUNE_ACL_CHECK in cdap-site.xml.